medicai/layers/pooling.py

```
@@ -0,0 +1,262 @@
+        **kwargs,
+    ):
+        self.reduce_function = reduce_function
+        self.output_size = ensure_tuple_rep(output_size, 3)
+        super().__init__(**kwargs)
+
+    def build(self, input_shape):
+        if len(input_shape) != 4:
+            raise ValueError(
+                f"{self.__class__.__name__} expects input with 4 dims (batch, H, W, C), "
+                f"but got {input_shape}"
+            )
```

Ruff (0.14.2)

Avoid specifying long messages outside the exception class

(TRY003)